dev/src/reference.ts

```diff
@@ -3051,76 +3052,127 @@ export class AggregateQuery<T extends firestore.AggregateSpec>
     return runQueryRequest;
   }
 
+  /**
+   * Compares this object with the given object for equality.
+   *
+   * This object is considered "equal" to the other object if and only if
+   * `other` performs the same aggregations as this `AggregateQuery` and
+   * the underlying `Query` of `other` compares equal to that of this object.
+   *
+   * @param other The object to compare to this object for equality.
+   * @return `true` if this object is "equal" to the given object, as
+   * defined above, or `false` otherwise.
+   */
   isEqual(other: firestore.AggregateQuery<T>): boolean {
     if (this === other) {
       return true;
     }
-    if (other instanceof AggregateQuery) {
-      if (!this._query.isEqual(other._query)) {
-        return false;
-      }
-
-      const thisAggregates: [string, AggregateField<unknown>][] =
-        Object.entries(this._aggregates);
-      const otherAggregates = other._aggregates;
+    if (!(other instanceof AggregateQuery)) {
+      return false;
+    }
+    if (!this.query.isEqual(other.query)) {
+      return false;
+    }
 
-      return (
-        thisAggregates.length === Object.keys(otherAggregates).length &&
-        thisAggregates.every(([alias, field]) =>
-          field.isEqual(otherAggregates[alias])
-        )
-      );
+    // Verify that this object has the same aggregation keys as `other`.
```

Comment:
Can we do this without sort? Maybe there exists or we can create a helper method?

The consequence is likely small since we are likely only dealing with maps of one element. So, maybe we can just have a TODO to revisit this when other aggregate types are implement.

https://stackoverflow.com/a/35951373

Reply:
I looked a little harder (I must have missed this while coding late last night!) and there already exists a `deepEqual()` function that employs the logic you suggested. So I've modified `AggregateQuery.isEqual()` and `AggregateQuerySnapshot.isEqual()` to just use `deepEqual()`.